CMakeLists.txt

```diff
@@ -218,6 +218,12 @@ append_if(CXX_SUPPORTS_UNREACHABLE_CODE_FLAG "-Wunreachable-code" CMAKE_CXX_FLAG
 check_cxx_compiler_flag("-Werror -Woverloaded-virtual" CXX_SUPPORTS_OVERLOADED_VIRTUAL)
 append_if(CXX_SUPPORTS_OVERLOADED_VIRTUAL "-Woverloaded-virtual" CMAKE_CXX_FLAGS)
 
+# Release configuration has assertions disabled, enable them if asked to.
+if(ENABLE_RELEASE_ASSERTIONS)
+  string(REPLACE "-DNDEBUG" "" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+  string(REPLACE "-DNDEBUG" "" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+endif()
```

Comment:
Let's call this `LLBUILD_ENABLE_ASSERTIONS` for consistency with `LLVM_ENABLE_ASSERTIONS`

Reply:
Also, based on LLVM I think you can just set `ENABLE_ASSERTIONS`
```
  if(LLVM_ENABLE_ASSERTIONS AND NOT MSVC_IDE)
    set(ENABLE_ASSERTIONS "1")
  else()
    set(ENABLE_ASSERTIONS "0")
  endif()
```

Reply:
Thats for the lit configuration. Unfortunately, there is no way to control assertions cleanly in CMake.